app/src/main/java/dcom/taneja/dhruv/floater/screenshotservice/ScreenshotActivity.java

```diff
@@ -0,0 +1,171 @@
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+
+import dcom.taneja.dhruv.floater.R;
+
+/**
+ * Created by PrateekGarg on 1/12/2018.
+ */
+
+public class ScreenshotActivity extends AppCompatActivity {
+    private static final int PERMISSION_REQUEST_EXTERNAL_STORAGE = 537;
+    private String mPathToFile;
+    private String TAG = ScreenshotActivity.class.getSimpleName();
+    private TextView mTextViewTimer;
+    private View mLayout;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.screenshot_activity);
+        mLayout = findViewById(R.id.main_activity);
+        findViewById(R.id.button_screenshot_action).setOnClickListener(new View.OnClickListener() {
+
+            @Override
+            public void onClick(View view) {
+                beginScreenshotTaker();
+            }
+        });
+    }
+    @Override
+    public void onRequestPermissionsResult(int requestCode, String[] permissions,
+                                           int[] grantResults) {
+        // BEGIN_INCLUDE(onRequestPermissionsResult)
+        if (requestCode == PERMISSION_REQUEST_EXTERNAL_STORAGE) {
+            // Request for write permission.
+            if (grantResults.length == 1 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                // Permission has been granted. Start screenshot preview Activity.
+                Snackbar.make(mLayout, "Write permission was granted. Starting preview.",
+                        Snackbar.LENGTH_SHORT)
+                        .show();
+                takeScreenshot();
+            } else {
+                // Permission request was denied.
+                Snackbar.make(mLayout, "Write permission request was denied.",
+                        Snackbar.LENGTH_SHORT)
+                        .show();
+            }
+        }
+        // END_INCLUDE(onRequestPermissionsResult)
+    }
+
+    private void beginScreenshotTaker() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            // BEGIN_INCLUDE(startScreenshot)
+            // Check if the Write permission has been granted
+            if (ActivityCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE)
+                    == PackageManager.PERMISSION_GRANTED) {
+                // Permission is already available, start screenshot
+                takeScreenshot();
+            } else {
+                // Permission is missing and must be requested.
+                requestExternalStoragePermission();
+            }
+            // END_INCLUDE(startScreenshot)
+        } else { //permission is automatically granted on sdk<23 upon installation
+            Log.v(TAG, "Permission is granted");
```

Comment:
Wherever you are putting logs for debugging, use Log.d
Log.d only spits out logs in debug mode.
If you are a library and want to log errors/warnings/etc, use other logging methods to give suitable logs.

Reply:
This is intentionally left to verbose. I don't want this message populated in prod.